src/types.rs

```diff
@@ -488,6 +490,146 @@ impl<T> Line<T>
+        }
+    }
+
+    /// Create a new `LineString` from a `Point`s vector. Only use this
+    /// function if you know _for certain_ that `points` has at least two
+    /// `Point`s.
+    ///
+    /// # Safety
+    ///
+    /// This function is `unsafe` because (soon) some `LineString` internals
+    /// ignore bounds checking when indexing as a performance improvement.
+    ///
+    /// # Examples
+    ///
+    /// Creating a `LineString`:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// let line_string = unsafe { LineString::new_unchecked(points) };
+    /// ```
+    pub unsafe fn new_unchecked(points: Vec<Point<T>>) -> Self {
+        debug_assert!(points.len() > 1);
+        LineString(points)
+    }
+
+    /// Retrieve a slice of this `LineString`’s underlying `Point`s.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// let line_string = LineString::new(points).unwrap();
+    ///
+    /// assert_eq!(line_string.points(), &[
+    ///     Point::new(6., 14.),
+    ///     Point::new(2., -10.),
+    /// ]);
+    /// ```
+    pub fn points(&self) -> &[Point<T>] {
+        &self.0
+    }
+
+    /// Retrieve a mutable slice of this `LineString`’s underlying `Point`s.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// let mut line_string = LineString::new(points).unwrap();
+    ///
+    /// for point in line_string.points_mut() {
+    ///     point.0.y *= 2.;
+    /// }
+    ///
+    /// assert_eq!(line_string.points(), &[
+    ///     Point::new(6., 28.),
+    ///     Point::new(2., -20.),
+    /// ]);
+    /// ```
+    pub fn points_mut(&mut self) -> &mut [Point<T>] {
+        &mut self.0
+    }
+
+    /// Determine if this `LineString` is ‘closed’, such that the first and last
+    /// `Point`s are equivalent.
+    ///
+    /// # Examples
+    ///
+    /// A `LineString` that is closed:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![
+    ///     Point::new(1., 5.),
+    ///     Point::new(2., -10.),
+    ///     Point::new(-4., 5.),
+    ///     Point::new(1., 5.),
+    /// ];
+    /// let line_string = LineString::new(points).unwrap();
+    /// assert!(line_string.is_closed());
+    /// ```
+    ///
+    /// A `LineString` that is not closed:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![
+    ///     Point::new(1., 5.),
+    ///     Point::new(2., -10.),
+    ///     Point::new(-4., 5.),
+    /// ];
+    /// let line_string = LineString::new(points).unwrap();
+    /// assert!(!line_string.is_closed());
+    /// ```
+    pub fn is_closed(&self) -> bool {
+        assert!(self.0.len() > 1);
```

Comment:
this assertion can be removed once we remove the `pub` in `LineString<T>(pub Vec<Point<T>>)`

Reply:
It might be useful to keep the function around and use it for validation in the constructors, but we can probably remove the `pub` from `pub fn is_closed()`. Users shouldn't need to worry about checking if a line is closed seeing as it's an explicit assumption upheld by the constructor.

Reply:
> Users shouldn't need to worry about checking if a line is closed seeing as it's an explicit assumption upheld by the constructor.

To clarify, this is an `is_closed` method on a _line string_ and not a _polygon_. I anticipate the polygon constructor in the future will ensure the interior data is a closed line string, but it could use this method.